cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -1203,7 +1203,7 @@ protected String determineImagePullPolicy(ImagePullPolicy requestedImagePullPoli
         }
     }
 
-    String getAncillaryConfigMapKeyLogConfig() {
+    public String getAncillaryConfigMapKeyLogConfig() {
```

Comment:
Maybe it would be better, at this point to change the value of `ANCILLARY_CM_KEY_LOG_CONFIG` to `"log4j2.properties"` and override `getAncillaryConfigMapKeyLogConfig()` on the Kafka and other things which are still log4j 1.

Reply:
Why? For making `log4j2` the new default?

Reply:
Yes, AFAIK all the Strimzi stuff is using log4j2 at this point. Kafka will move at some point. IDK about Zookeeper, but I'm not aware of any reason it has to be log4j 1 any more. So it makes more sense to just override this method for Kafka than having to override it for everything except Kafka.

Reply:
I searched a little bit more in the code and it seems we have ZooKeeper, Kafka, Kafka Connect, Cruise Control, Mirror Maker and Mirror Maker 2 all using log4j 1 ... while just operator and bridge are using log4j2. So 6 vs 3 ... is it really useful doing the change now?

Reply:
No, I guess not. 

However, we're going to have to do this at some point:

* log4jv1 has at least 1 CVE and will not be fixed, so there are good reasons to move
* Zookeeper and CC both use slf4j as a facade, so there's a good chance we could switch to using lo4j2 as the back end with no ill effects. It should be enough to grep to see if they `import` anything more than loggers.
* https://github.com/apache/kafka/pull/7898 is making steady progress, so hopefully it won't be long before Kafka changes too.

So maybe I'm a little premature, but our work here is not done.

Reply:
We can open an issue and propose the change to the Cruise Control community? Do you think that we should contribute to the same for ZooKeeper as well? Anyway, we rely on the ZooKeeper version brought by Kafka, right? So even if moving to log4j2 on Zookeeper we have to wait for Kafka community to takes that version.

Reply:
See https://issues.apache.org/jira/browse/ZOOKEEPER-2342. The comments suggest that is basically just works, and the only reason they've stuck with v1 so far is simply for compatibility with v1 configuration syntax. Also:

```
$ grep -Fr 'import org.apache.log' .
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/Log4jAuditLoggerTest.java:import org.apache.log4j.Layout;
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/Log4jAuditLoggerTest.java:import org.apache.log4j.Level;
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/Log4jAuditLoggerTest.java:import org.apache.log4j.Logger;
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/Log4jAuditLoggerTest.java:import org.apache.log4j.SimpleLayout;
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/Log4jAuditLoggerTest.java:import org.apache.log4j.WriterAppender;
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/StandaloneServerAuditTest.java:import org.apache.log4j.Layout;
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/StandaloneServerAuditTest.java:import org.apache.log4j.Level;
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/StandaloneServerAuditTest.java:import org.apache.log4j.Logger;
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/StandaloneServerAuditTest.java:import org.apache.log4j.SimpleLayout;
./zookeeper-server/src/test/java/org/apache/zookeeper/audit/StandaloneServerAuditTest.java:import org.apache.log4j.WriterAppender;
./zookeeper-server/src/test/java/org/apache/zookeeper/test/ReadOnlyModeTest.java:import org.apache.log4j.Layout;
./zookeeper-server/src/test/java/org/apache/zookeeper/test/ReadOnlyModeTest.java:import org.apache.log4j.Level;
./zookeeper-server/src/test/java/org/apache/zookeeper/test/ReadOnlyModeTest.java:import org.apache.log4j.Logger;
./zookeeper-server/src/test/java/org/apache/zookeeper/test/ReadOnlyModeTest.java:import org.apache.log4j.WriterAppender;
./zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java:import org.apache.log4j.Level;
./zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java:import org.apache.log4j.Logger;
./zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java:import org.apache.log4j.PatternLayout;
./zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java:import org.apache.log4j.WriterAppender;
```

I.e. the only place where they depend on log4j v1 is in the tests.

For CC that grep shows nothing. 

So there's pretty much nothing to do, unless you want to get bogged down in the config compatibility debate. But that needn't concern us: In both cases we could switch really easily if we wanted to, it's just a matter of writing a log4j2 config file or using v2's support for reading v1 configs: http://logging.apache.org/log4j/2.x/manual/migration.html and arranging the classpath appropriately.